src/components/body/hooks/useTheme.tsx

```diff
@@ -0,0 +1,11 @@
+import { useLayoutEffect, useState } from "react";
+
+export const useTheme=()=>{
+    const [theme, setTheme]=useState(localStorage.getItem("selectTheme") || "dark")
+    useLayoutEffect(()=>{
+        document.documentElement.setAttribute("data-theme", theme)
+        localStorage.setItem("selectTheme", theme)
+    }, [ theme ]);
+
+    return {theme,setTheme}
```

Comment:
it is better to return an array here as it pretends to custom hook. Hooks always return arrays. Try to refactor...